src/lib/merge-selectors.js

```diff
@@ -1,14 +1,27 @@
-import { complex, replaceable } from './valid-selector.js'
+import { replaceable } from './valid-selector.js'
 
 export default function mergeSelectors(fromSelectors, toSelectors) {
-	return fromSelectors.reduce(
-		(selectors, fromSelector) => selectors.concat(
-			toSelectors.map(
-				(toSelector) => complex.test(toSelector)
-					? toSelector.replace(replaceable, `:is(${fromSelector})`)
-				: toSelector.replace(replaceable, fromSelector)
-			)
-		),
-		[]
-	)
+	return toSelectors.map((toSelector) => {
+		let needsIsOnFromSelector = false
+
+		if (fromSelectors.length > 1) {
+			needsIsOnFromSelector = true
+		}
+
+		// foo &foo foo & baz -> foo &:is(foo) foo & baz
+		toSelector = toSelector.replace(/&((?:[\w-_|])(?:[^\s,{]*))/g, (match, p1) => {
```

Comment:
If we’re open to regular expressions, then we should use a tokenizer. I will try to make a PR to this branch today. I expect to start on the work in about 5 hours.

Reply:
@jonathantneal Maybe with `postcss-selector-parser`?

https://github.com/romainmenke/postcss-nesting/pull/1

Might be too big a hammer, but could be easier to maintain than another tokenizer :)

Reply:
I was off by several hours, because I forgot it’s the day we all hand out treats to the neighborhood kids!